### ISSUE TITLE
2^256 == 0, 2^255 == -578...

### ISSUE THREAD

**Issue by davebryson** [ethereum/serpent#95]
In the wiki docs for Serpent, it says that anything over 2^256 will wrap to 1.  Using serpent 2.0.2 I'm finding that:

`2^256 = 0`
`2^255 = -5789604461... (note that's  negative) `

And anything over 2^256 = 0

Is this correct behavior or a bug?

**Comment by joeykrug** [comment]
That's correct behavior.  Serpent uses signed integers --- see http://kias.dyndns.org/comath/13.html "if a signed integer has n bits, it can contain a number between -2^(n - 1) and 2^(n - 1) - 1"
